### PR TITLE
Move account creation of scheduled users to the first day

### LINKED ISSUE #989: Move account creation of scheduled users to the first day
Do not create it after confirmation, rather schedule to the first day

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #989 

## PR includes

-   [x] Feature

### CODE REVIEW

File: src/modules/historyEventTypes.ts [hunk 1]
```diff
@@ -64,10 +64,12 @@ interface HistoryEventsData {
     };
     acceptUserCreationRequest: {
         requireGroupId: false;
-        requireUserId: true;
+        requireUserId: false;
         requireBefore: false;
         data: {
             id: string;
+            name: string;
```
Comment (tsumo) [inline src/modules/historyEventTypes.ts:71]: new fields must be optional, they don't exist in previous logged events

File: src/modules/userCreationRequestMethods.ts [hunk 2]
```diff
@@ -201,55 +211,20 @@ export const userCreationRequestsMethods = {
...
-                login: acceptedRequest.login,
-                title: userCreationRequest.title,
-                memberships: { create: { groupId: acceptedRequest.groupId } },
-                organizationUnitId: acceptedRequest.organizationUnitId,
-                services: { createMany: { data: services } },
-                workStartDate: acceptedRequest.date,
-            },
-        });
+        if (acceptedRequest.date && new Date() > acceptedRequest.date) {
+            console.log('accd', acceptedRequest.date);
```
Comment (tsumo) [inline src/modules/userCreationRequestMethods.ts:223]: logs

File: src/modules/userCreationRequestMethods.ts [hunk 3]
```diff
@@ -201,55 +211,20 @@ export const userCreationRequestsMethods = {
...
+            console.log('newdara', new Date());
+            console.log(new Date() > acceptedRequest.date);
+            newUser = await userMethods.createUserFromRequest(acceptedRequest.id);
```
Comment (tsumo) [inline src/modules/userCreationRequestMethods.ts:226]: I think user creation should always be through the job

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/modules/userCreationRequestMethods.ts (hunk 3 above)
Comment (nyet-ty) [reply src/modules/userCreationRequestMethods.ts:226]: Its if request accepted after job already done and deleted. Do you suggest to delay job, if request not accepted in time?

**Review by Katrin-kudryash — APPROVED** [review]

File: src/trpc/router/userCreationRequestRouter.ts [hunk 4]
```diff
@@ -70,19 +70,21 @@ export const userCreationRequestRouter = router({
...
             after: {
                 id: acceptedRequest.id,
+                name: acceptedRequest.name,
+                email: acceptedRequest.email,
                 comment: input.comment,
             },
         });
 
-        return { newUser, acceptedRequest };
+        acceptedRequest;
```
Comment (tsumo) [inline src/trpc/router/userCreationRequestRouter.ts:87]: return

File: src/modules/userMethods.ts [hunk 5]
```diff
@@ -502,4 +502,63 @@ export const userMethods = {
 
         return countUserLogins + countUserRequestLogins === 0;
     },
+    createUserFromRequest: async (userCreationRequestId: string) => {
```
Comment (tsumo) [inline src/modules/userMethods.ts:505]: empty line between methods

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/modules/userMethods.ts (hunk 5 above)
Comment (nyet-ty) [reply src/modules/userMethods.ts:506]: done

**Review by tsumo — APPROVED** [review]

**Pull request merged**